client/src/main/kotlin/io/provenance/classification/asset/client/domain/execute/AddAssetDefinitionExecute.kt

```diff
@@ -1,104 +1,32 @@
 
-import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy
-import com.fasterxml.jackson.databind.annotation.JsonNaming
+import com.fasterxml.jackson.databind.annotation.JsonSerialize
 import io.provenance.classification.asset.client.domain.execute.base.ContractExecute
 import io.provenance.classification.asset.client.domain.model.ScopeSpecIdentifier
 import io.provenance.classification.asset.client.domain.model.VerifierDetail
-import java.util.UUID
+import io.provenance.classification.asset.client.domain.serialization.AddAssetDefinitionExecuteSerializer
 
 /**
- * This class is a reflection of the request body used in the Asset Classification smart contract's add or update asset
- * definition execution routes.
+ * This class is a reflection of the request body used in the Asset Classification smart contract's add asset
+ * definition execution route.
  *
- * This request body is relatively complex, and re-uses its internal body across two calls (add/update).
- * To use it, simply create the execute class, and then choose add or update:
+ * To use it, simply create the execute class and call the appropriate function:
  * ```kotlin
- * val execute = AssetDefinitionExecute.withAssetUuid(uuid, assetType, verifiers, enabled = true)
- * // An add will only execute correctly if it is a brand new asset definition
- * val addRequest = execute.toAdd()
- * val txResponse = acClient.addAssetDefinition(addRequest, signer, options)
- * // An update will only execute correctly if it is an existing asset definition
- * val updateRequest = execute.toUpdate()
- * val txResponse = acClient.updateAssetDefinition(updateRequest, signer, options)
+ * val execute = AddAssetDefinitionExecute(assetType, ScopeSpecIdentifier.Uuid(UUID.randomUUID()), verifiers, enabled = true)
+ * val txResponse = acClient.addAssetDefinition(execute, signer, options)
  * ```
  *
- * @param assetType The type of asset that will be added or updated. This value is a unique key in the contract.
+ * @param assetType The type of asset that will be added. This value is a unique key in the contract.
  * @param scopeSpecIdentifier Identifies the scope spec that this asset definition is associated with. This value is a unique constraint and can only be mapped to one asset definition.
  * @param verifiers All verifiers that are allowed to do verification for this specific asset type.
  * @param enabled Whether or not this asset type will accept incoming onboard requests.  If left null, the default value used will be `true`
- * @param bindName Whether or not to bind the name value creating an asset definition. Only affects the result when being used with the "add" functionality.
+ * @param bindName Whether or not to bind the name value creating an asset definition.
  */
-@JsonNaming(SnakeCaseStrategy::class)
-data class AssetDefinitionExecute<T> internal constructor(
+@JsonSerialize(using = AddAssetDefinitionExecuteSerializer::class)
+data class AddAssetDefinitionExecute<T>(
```

Comment:
I moved this structure out to two individual data classes with their own files `AddAssetDefinitionExecute` and `UpdateAssetDefinitionExecute`, which both use custom serializers to remain a flat class structure while including multiple wrapper nodes.